ObjectiveCloudant/Operations/CDTQueryViewOperation.m

```diff
@@ -0,0 +1,291 @@
+                                      value:[NSString stringWithFormat:@"%ld", (long)self.limit]]];
+    }
+
+    [queryItems addObject:[NSURLQueryItem queryItemWithName:@"reduce"
+                                                      value:[CDTQueryViewOperation
+                                                                stringForBoolean:self.reduce]]];
+
+    if (self.skip != kCDTDefaultOperationIntegerValue) {
+        [queryItems
+            addObject:[NSURLQueryItem
+                          queryItemWithName:@"skip"
+                                      value:[NSString stringWithFormat:@"%ld", (long)self.skip]]];
+    }
+
+    NSURLQueryItem *stale = [CDTQueryViewOperation QueryItemForStaleState:self.stale];
+
+    if (stale) {
+        [queryItems addObject:stale];
+    }
+
+    if (self.startKey) {
+        [queryItems
+            addObject:[NSURLQueryItem
+                          queryItemWithName:@"startkey"
+                                      value:[CDTQueryViewOperation stringForObject:self.startKey]]];
+    }
+
+    if (self.startKeyDocId) {
+        [queryItems addObject:[NSURLQueryItem queryItemWithName:@"startkey_docid"
+                                                          value:self.startKeyDocId]];
+    }
+
+    return [queryItems copy];
+}
+
+- (void)callCompletionHandlerWithError:(NSError *)error
+{
+    if (self && self.queryViewCompletionBlock) {
+        self.queryViewCompletionBlock(error);
+    }
+}
+
+- (void)processResponseWithData:(NSData *)responseData
+                     statusCode:(NSInteger)statusCode
+                          error:(NSError *)error
+{
+    if (error) {
+        self.queryViewCompletionBlock(error);
+    } else if (statusCode / 100 == 2) {
+        // process the things!
+        NSError *jsonError;
+        NSDictionary<NSString *, NSObject *> *responseJson =
+            [NSJSONSerialization JSONObjectWithData:responseData options:0 error:&jsonError];
+
+        if (!responseJson) {
+            self.queryViewCompletionBlock(jsonError);
+        } else {
+            if (self.viewRowBlock) {
+                for (NSDictionary<NSString *, NSObject *> *row in(
+                         NSArray<NSDictionary<NSString *, NSObject *> *> *)responseJson[@"rows"]) {
+                    self.viewRowBlock(row);
+                }
+            }
+            if (self.queryViewCompletionBlock) {
+                self.queryViewCompletionBlock(nil);
+            }
+        }
+
+    } else {
+        NSString *json = [[NSString alloc] initWithData:responseData encoding:NSUTF8StringEncoding];
+        NSString *msg = [NSString
+            stringWithFormat:@"Querying view failed with %ld %@.", (long)statusCode, json];
+        NSDictionary *userInfo = @{NSLocalizedDescriptionKey : NSLocalizedString(msg, nil)};
+        error = [NSError errorWithDomain:CDTObjectiveCloudantErrorDomain
+                                    code:CDTObjectiveCloudantErrorCreateDatabaseFailed
```

Comment:
We should probably add an error code specific to a view query failure and use it here.